server_development/topics/auth-spi.adoc

```diff
@@ -152,78 +152,204 @@ org.keycloak.examples.authenticator.AnotherProviderFactory
+from the negotiate header. This example, however, is validating a specific credential of a specific user.
+
+[source,java]
+----
+    @Override
+    public boolean requiresUser() {
+        return true;
+    }
+----
 
 The next method to implement is the configuredFor() method.
 This method is responsible for determining if the user is configured for this particular authenticator.
 For this example, we need to check if the answer to the secret question has been set up by the user or not.
-In our case we are storing this information, hashed, within a UserCredentialValueModel within the UserModel (just like passwords are stored).  Here's how we do this very simple check: 
+In our case we are storing this information, hashed, within a UserCredentialValueModel within the UserModel
+(just like passwords are stored).  Here's how we do this very simple check:
 
 [source,java]
 ----
-@Override
-  public boolean configuredFor(KeycloakSession session, RealmModel realm, UserModel user) {
-     return session.users().configuredForCredentialType("secret_question", realm, user);
+    /**
+     * The type string separates this Authenticator from other Authenticators.
+     */
+    public static final String SECRET_QUESTION_TYPE = "secret_question";
+
+    @Override
+    public boolean configuredFor(final KeycloakSession session,
+                                 final RealmModel realm,
+                                 final UserModel user) {
+
+        return session.userCredentialManager().isConfiguredFor(realm, user, SECRET_QUESTION_TYPE);
     }
 ----            
 
-The configuredForCredentialType() call queries the user to see if it supports that credential type. 
+The isConfiguredFor() call queries the user to see if it supports that credential type.
 
-The next method to implement on the Authenticator is setRequiredActions().  If configuredFor() returns false and our example authenticator is required within the flow, this method will be called.
+The next method to implement on the Authenticator is setRequiredActions().
+If configuredFor() returns false and our example authenticator is required within the flow, this method will be called.
 It is responsible for registering any required actions that must be performed by the user.
 In our example, we need to register a required action that will force the user to set up the answer to the secret question.
 We will implement this required action provider later in this chapter.
 Here is the implementation of the setRequiredActions() method. 
 
 [source,java]
 ----
+    /**
+     * Called if this Authenticator is <em>not</em> configured for the provided user (i.e. if the
+     * {@link #configuredFor(KeycloakSession, RealmModel, UserModel)} method returned <code>false</code>).
+     * This method defines required actions which must be performed by the user before this {@link Authenticator}
+     * can be used.
+     *
+     * {@inheritDoc}
```

Comment:
The build seems to treat the inheritDoc as a variable rather than as part of the javadoc.
Simply remove this line and we should be OK to merge - this is what the build log complains about.

```suggestion 
```